Source/WTF/wtf/SuspendableWorkQueue.cpp

```diff
@@ -26,24 +26,43 @@
 #include "config.h"
 #include <wtf/SuspendableWorkQueue.h>
 
+#include <wtf/Logging.h>
+
 namespace WTF {
 
-Ref<SuspendableWorkQueue> SuspendableWorkQueue::create(const char* name, WorkQueue::QOS qos)
+Ref<SuspendableWorkQueue> SuspendableWorkQueue::create(const char* name, WorkQueue::QOS qos, ShouldLog shouldLog)
 {
-    return adoptRef(*new SuspendableWorkQueue(name, qos));
+    return adoptRef(*new SuspendableWorkQueue(name, qos, shouldLog));
 }
 
-SuspendableWorkQueue::SuspendableWorkQueue(const char* name, QOS qos)
+SuspendableWorkQueue::SuspendableWorkQueue(const char* name, QOS qos, ShouldLog shouldLog)
     : WorkQueue(name, qos)
+    , m_shouldLog(shouldLog == ShouldLog::Yes)
 {
     ASSERT(isMainThread());
 }
 
+const char* SuspendableWorkQueue::stateString(State state)
```

Comment:
I'd recommend to define this only in the cpp (since you don't need outside this class and mark as inline:
```
static inline const char* SuspendableWorkQueue::stateString(State state)
{
...
}
```

Just to be extra sure we don't generate a function call.